### PR TITLE
Add some support for Undertow instances bound to DI lifecycles

### PR DESCRIPTION
**Pull request by twz123**
When using a DI context that supports lifecycles, like Guice in conjunction with Governator, one wants to have some support for letting the DI container manage Undertow instances.

### CODE REVIEW

File: src/main/java/org/zalando/undertaking/inject/GracefulShutdown.java [hunk 1]
```diff
@@ -0,0 +1,58 @@
...
+import io.undertow.server.HttpHandler;
+import io.undertow.server.handlers.GracefulShutdownHandler;
+
+/**
+ * Wraps {@code HttpHandlers} in order to provide graceful shutdown behavior. May be used as a in DI contexts for
+ * lifecycle management.
+ */
+public class GracefulShutdown implements HandlerWrapper, AutoCloseable {
+
+    private Collection<GracefulShutdownHandler> handlers = new ArrayDeque<>(4);
```
Comment (fatroom) [inline src/main/java/org/zalando/undertaking/inject/GracefulShutdown.java]: Why 4?

**Comment by fatroom** [comment]
👍

**Comment by twz123** [comment]
I'll add some tests today...

**Comment by twz123** [comment]
:+1:

**Comment by cberg-zalando** [comment]
:+1:

**Pull request merged**